SDL/Kernels.h

```diff
@@ -479,26 +479,26 @@ namespace SDL
                 float eta_pix1 = segmentsInGPU.eta[ix];
                 float phi_pix1 = segmentsInGPU.phi[ix];
 
-                for(int jx = globalThreadIdx[2]; jx < nPixelSegments; jx += gridThreadExtent[2])
+                for(int jx = ix + 1 + globalThreadIdx[2]; jx < nPixelSegments; jx += gridThreadExtent[2])
                 {
                     float eta_pix2 = segmentsInGPU.eta[jx];
                     float phi_pix2 = segmentsInGPU.phi[jx];
 
-                    if (ix == jx)
-                        continue;
-
                     if (alpaka::math::abs(acc, eta_pix2 - eta_pix1) > 0.1f)
                         continue;
 
                     if (secondpass && (!segmentsInGPU.isQuad[jx] || segmentsInGPU.isDup[jx]))
                         continue;
```

Comment:
I'm also curious about the original code here, why do we think that, if a pLS is not quad, (which means it's a triplet), or it is duplicated, we continue and don't remove it?

Reply:
That's a good question. I'm also not sure why that's done. We should ask someone more familiar with this part of the code.

Reply:
@YonsiG We skip adding the triplet and duplicate pLS in the TC candidates (in the `addpLSasTrackCandidateInGPU` kernel), so I believe the logic is to not waste time checking objects that are not going to be used in any case.

In the meeting, the idea to mark all of the objects to be skipped with a single variable, instead of having multiple flags in multiple places. I will make an issue out of that, so that we keep it in mind.

Reply:
Thank you for explaining Manos!